### ISSUE TITLE
[📚] Migrating FlutterFire initialization misleading docs

### ISSUE THREAD

**Issue by edhom** [FirebaseExtended/flutterfire#7866]
This tip is really misleading. You cannot safely remove the configuration files as long as there are services not supporting dart initialization.
<img width="976" alt="Screen Shot 2022-01-14 at 09 48 08" src="https://user-images.githubusercontent.com/47747733/149493817-6a54a8c4-5ab4-4b6c-8412-262de762c982.png">

**Comment by darshankawar** [comment]
> as long as there are services not supporting dart initialization.

See if this link helps to answer which services are supported and which are not : https://firebase.flutter.dev/docs/cli/#usage

**Comment by pree-T** [comment]
May I work on it? Can someone help me as I am new to open source contribution?

**Comment by edhom** [comment]
@darshankawar Thanks for the link👍 So in case you'd like to migrate, but use one or more of these unsupported services, can you use the old configuration files to initialize those and dart initialization for everything else? Or in other words: Can you migrate partially? If so, is there anything you need to pay attention to? (E.g. adjusting the configuration files?). Would be really thankful for some advise here!

**Comment by creativecreatorormaybenot** [comment]
@darshankawar That would mean that [the docs](https://firebase.flutter.dev/docs/overview#initializing-flutterfire) are wrong:

> If you're using FlutterFire in an existing application, you can safely remove any manual native installation steps (such as the google-services.json or GoogleService-Info.plist files). Dart-only initialization is backwards compatible with manual-installation.

This is not in accordance with the link you shared (https://firebase.flutter.dev/docs/cli/#usage) because for Analytics, Crashlytics, Dynamic Links, and Performance Monitoring, the files are required.

**Comment by darshankawar** [comment]
> Can you migrate partially? If so, is there anything you need to pay attention to? (E.g. adjusting the configuration files?). Would be really thankful for some advise here!

I haven't tried to implement partial migration yet, so not sure how it works currently. Keeping it open and labeling for team to provide input on such scenario.

/cc @russellwheatley

**Comment by neiljaywarner** [comment]
@darshankawar @russellwheatley @creativecreatorormaybenot  I love flutter and firebase but this made me very sad.  analytics and crashlytics are the two most key features. I'm also using remote config. 
The big green box saying "no longer recommended" and i think reference to having changed this in mid 2020.  Not working with android on crashlytics (there's a non-clickable question mark and it doens't work for me, gives analytics disabled, no google app id but works on ios).
basically the whole idea seems practically broken, it seemed like such a great idea too, especially when using flavors...

Can you please escalate? @puf or anybody . if you can't use analytics on android then it can't be teh "main" recommended way.
`E/FA      (27683): Missing google_app_id. Firebase Analytics disabled. See https://goo.gl/NAOOOI
`

**Comment by neiljaywarner** [comment]
@russellwheatley @darshankawar  this is good but what about an eta on the real problem? Getting this working would make flavors much less painful i think :)

**Comment by russellwheatley** [comment]
@neiljaywarner - we're working on it. Some of the work required is out of our hands I'm afraid so we cannot give you a timeframe, but we'll update the documentation as and when the landscape changes.

**Comment by neiljaywarner** [comment]
Thank you

....sent from my phone
